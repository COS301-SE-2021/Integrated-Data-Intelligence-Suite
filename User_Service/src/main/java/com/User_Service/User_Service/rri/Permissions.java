package com.User_Service.User_Service.rri;

public enum Permissions {
    IMPORTING,
    VIEWING
}
