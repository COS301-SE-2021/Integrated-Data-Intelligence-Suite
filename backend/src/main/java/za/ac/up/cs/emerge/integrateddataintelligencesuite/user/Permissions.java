package za.ac.up.cs.emerge.integrateddataintelligencesuite.user;

public enum Permissions {
    USER,
    ADMIN
}

