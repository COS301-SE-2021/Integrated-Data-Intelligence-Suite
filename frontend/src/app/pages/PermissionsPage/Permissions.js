import UserList from "../../components/UserList/UserList";
import SideBar from "../../components/SideBar/SideBar";
import {Divider, Layout, Typography} from "antd";
import useGet from "../../functions/useGet";
import React, {Component, useRef} from 'react';

const {Content, Footer, Header} = Layout;
const {Title} = Typography;


const Permissions = () => {
    const {data: users, isPending, error} = useGet('/user/getAll');

    return (
        <Layout>
            <Content className={"permissions-content-section"}>
                <div className={"permissions user"}>
                    {users && <UserList users={users.users}/>}
                </div>
            </Content>
        </Layout>
    );
};

export default Permissions;
