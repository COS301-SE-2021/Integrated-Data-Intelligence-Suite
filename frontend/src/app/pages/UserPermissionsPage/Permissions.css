.user-preview{
    padding: 20px;
    margin-bottom: 10px;
    /*border-radius: 5px;*/
    line-height: normal;
    display: flex;
    background-color: white;
    border-radius: 10px;
    border: 1px solid #eff0f0;
}

.user-preview:hover{
    /*background-color: #5873f9;*/
    /*color: white;*/
    border-color: #5773FA;
}

.user-preview h3{
    font-size: 20px;
    font-weight: 600;
}

.user-list a{
    color: black;
    font-weight: 500;
}


.user-preview .admin-tag{
    color: #5773FA;
}

.user-preview a{
    /*padding: 0;*/
    display: flex;
    align-items: center;
    /*margin: 0 10px;*/
    min-height: 50px;
    text-decoration: none;
}
.user-preview p{
    margin-right: 0;
    margin-left: auto;
    display: flex;
}
.user-preview p select{
    margin: 10px;
    background: #5773FA;
    border: none;
    border-radius: 5px;
}

.permissions-content-section{
    padding: 20px;
    line-height: normal;

}
.user-info {
    /*display: flex;*/
    /*background:rgba(0,0,0,0);*/
    /*width: 50%;*/
    /*margin: auto;*/
    /*border-radius: 10px;*/
    /*padding: 50px;*/
    /*min-width: 350px;*/

    position: absolute;
    width: 24%;
    left: 50%;
    top: 50%;
    -webkit-transform: translate(-50%, -50%);
    transform: translate(-50%, -50%);
    padding: 20px;
}

.user-info h2{
    font-size: 25px;
    padding-bottom: 20px;
}

.user-info  p{
    padding-bottom: 10px;
}

.user-info form{
    border: none;
}
.bodyDiv .ant-layout{
    background-color: rgba(0,0,0,0);
}


.bodyDiv .ant-layout-content{
    /*background-color: lightgrey;*/
}

.bodyDiv .right-column .ant-layout-content{
    background-color: rgba(0,0,0,0);
}

.ant-divider#lightgrey{
    border-color: lightgray;
}


.clear-background h1#permissions-title{
    font-size: 28px;
    margin: 0;
    padding-left: 40px;
    text-transform: uppercase;
    font-weight: 600;


}


.bodyDiv .outer-container .left-column>div{
    color: black;
    text-transform: capitalize;
    font-size: 16px;
}


.form-container{

    border: 3px solid #5773FA;
    padding: 20px;
}




.user-info .form-button-container{
    display: inline-block;
    width: 100%;
    align-items: center;
}

.user-info form button{
    width: 200px;
}

.user-info form label{
    font-size: 14px;
    color: black;

}

.user-info form input[type="text"]{
    background: none;
    font-size: 14px;
    border: 1px solid lightgray;
    border-radius: 5px;
    width: 100%;
    padding: 5px;
    margin: 5px;
}

.user-info form select{
    background: none;
    border: 1px solid lightgray;
    border-radius: 5px;
    height: 35px;
    width: 100%;
    margin-bottom: 30px;
    margin-top: 10px;
    padding-left: 10px;

}

button.enabled{
    background: #5773FA;
    border: none;
    border-radius: 5px;
}

.form-header{
    display: flex;
}

.form-header span{
    float: left;
    margin-left: 150px;
    font-size: 20px;
}

.profile.user{
    margin: 20px;
}

.profile.user h2{

    padding: 0 40px 40px 40px;
    text-transform: uppercase;
    font-weight: 600;
    border: 1px solid #5773FA;
    line-height: normal;

}

.profile.user>div{
    line-height: normal;
    /*background: white;*/

}

.profile.user .about-user div{
    color: black;
}


