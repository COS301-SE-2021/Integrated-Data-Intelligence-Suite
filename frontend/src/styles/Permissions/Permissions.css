.user-preview{
    padding: 20px;
    margin-bottom: 10px;
    /*border-radius: 5px;*/
    line-height: normal;
    display: flex;
    background-color: lightgray;
    border-radius: 10px;
}

.user-preview:hover{
    background-color: #bfbebe;
}

.user-preview h3{
    font-size: 20px;
    font-weight: 600;
}

.user-list a{
    color: black;
    font-weight: 500;
}


.user-preview .admin-tag{
    color: #f1356d;
}

.user-preview a{
    /*padding: 0;*/
    display: flex;
    align-items: center;
    /*margin: 0 10px;*/
    min-height: 50px;
    text-decoration: none;
}
.user-preview p{
    margin-right: 0;
    margin-left: auto;
    display: flex;
}
.user-preview p select{
    margin: 10px;
    background: #15B761;
    border: none;
    border-radius: 5px;
}

.permissions-content-section{
    padding: 20px;
    line-height: normal;

}
.user-info {
    background:rgba(0,0,0,0);
    width: 50%;
    margin: 0 auto;
    border-radius: 10px;
    padding: 50px;
    align-items: center;
    min-width: 350px;
}

.user-info h2{
    font-size: 25px;
    padding-bottom: 20px;
}

.user-info  p{
    padding-bottom: 10px;
}

.user-info form{
    border: none;
}
.bodyDiv .ant-layout{
    background-color: rgba(0,0,0,0);
}


.bodyDiv .ant-layout-content{
    background-color: lightgrey;
}

.bodyDiv .right-column .ant-layout-content{
    background-color: rgba(0,0,0,0);
}

.ant-divider#lightgrey{
    border-color: lightgray;
}


.clear-background h1#permissions-title{
    font-size: 28px;
    margin: 0;
    padding-left: 40px;
    text-transform: uppercase;
    font-weight: 600;


}


.bodyDiv .outer-container .left-column>div{
    color: black;
    text-transform: capitalize;
    font-size: 16px;
}





.user-info .form-button-container{
    display: inline-block;
    width: 100%;
    align-items: center;
}

.user-info form button{
    width: 200px;
}

.user-info form label{
    font-size: 14px;
    color: black;

}

.user-info form input[type="text"]{
    background: #04AA6D;
    font-size: 14px;
}

.user-info form select{
    background: #04AA6D;
    border-radius: 2px;
    height: 35px;
    width: 100%;
    margin-bottom: 30px;
    margin-top: 10px;
    padding-left: 10px;

}

button.disabled{

}

