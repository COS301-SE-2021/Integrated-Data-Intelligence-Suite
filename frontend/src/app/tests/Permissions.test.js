import React from 'react';
import renderer from 'react-test-renderer';
import Permissions from "../pages/PermissionsPage/Permissions";

it('renders correctly', () => {
    const tree = renderer
        .create(<Permissions/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});