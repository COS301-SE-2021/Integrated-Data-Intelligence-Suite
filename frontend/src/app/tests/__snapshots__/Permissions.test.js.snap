// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  className="ant-layout"
>
  <main
    className="ant-layout-content permissions-content-section"
  >
    <div
      className="permissions user"
    />
  </main>
</section>
`;
